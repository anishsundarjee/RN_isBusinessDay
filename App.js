import 'react-native-gesture-handler';
import React from 'react'
import { View, Platform, StyleSheet, StatusBar } from 'react-native';
import FlashMessage from "react-native-flash-message";
import Colors from './src/constants/Colors'
import AppNavigator from './src/navigation/AppNavigator'

const App = () => {
	return (
		<View style={styles.container}>
			{Platform.OS === 'ios' && <StatusBar barStyle="default" />}
			<AppNavigator/>
			<FlashMessage position="top" /> 
		</View>
	)
};

export default App;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: Colors.PRIMARY_BG_COLOR,
	},
});
