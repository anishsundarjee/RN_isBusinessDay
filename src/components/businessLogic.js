import React from 'react'
// import { showMessage } from "react-native-flash-message";
import axios from 'axios';

const ENDPOINT = 'https://date.nager.at/api/v3/publicholidays/'
const COUNTRY = 'AU'

export const isWeekend = (input) => {
    try {
        if(isNaN(input.getTime())) {
            return undefined;
        }
        let checkDate = new Date(input);
        if(checkDate.getDay() === 6) { //if saturday, set to monday
            checkDate = checkDate.setDate(checkDate.getDate() + 2);
        }
        else if(checkDate.getDay() === 0) { //if sunday, set to monday
            checkDate = checkDate.setDate(checkDate.getDate() + 1);
        }
        else { //all good
            console.log('check :',checkDate);
        }
        return new Date(checkDate);
    } catch (error) {
        console.log('Error occured checking date for weekend',error)
        return error;
    }
}

export const addToDate = (input) => {
    try {
        if(isNaN(input.getTime())) {
            return undefined;
        }
        let checkDate = new Date(input);
        if(checkDate.getDay() === 6) { //saturday
            checkDate = checkDate.setDate(checkDate.getDate() + 2);
        }
        else if(checkDate.getDay() === 5) { //friday
            checkDate = checkDate.setDate(checkDate.getDate() + 3);
        }
        else { //other weekdays or sunday
            checkDate = checkDate.setDate(checkDate.getDate() + 1);
        }
        return new Date(checkDate);
    } catch (error) {
        console.log('Error occured adding to date if public holiday',error)
        return error;
    }
}

export const _setDateinFormat = (input) => {
    try {
        if(isNaN(input.getTime())) {
            return undefined;
        }
        let currentDayOfMonth = input.getDate();
        currentDayOfMonth < 9 ? currentDayOfMonth = `0${currentDayOfMonth}` : currentDayOfMonth

        let currentMonth = input.getMonth() + 1; // note! January is 0, not 1
        currentMonth < 10 ? currentMonth = `0${currentMonth}` : currentMonth

        const currentYear = input.getFullYear();

        return new Date(currentYear + "-" + currentMonth + "-" + currentDayOfMonth);
    } catch (error) {
        return error;
    }
}

export const _setOutputDateFormat = (input) => {
    try {
        if(isNaN(input.getTime())) {
            return undefined;
        }
        const currentDayOfMonth = input.getDate();
        const currentMonth = input.getMonth() + 1; // note! January is 0, not 1
        const currentYear = input.getFullYear();
        return currentYear + "-" + currentMonth + "-" + currentDayOfMonth
    } catch (error) {
        return error;
    }
}

export const _getListOfHolidays = async (input) => {
    try {
        if(isNaN(input)) {
            return undefined;
        }
        let res = null;
        const URL = `${ENDPOINT}${input}/${COUNTRY}`;
        await axios.get(URL,{ timeout:5000,
            headers : {
            'Accept': 'application/json',
            'Content-Type' : 'application/json'
        }})
        .then(response => {
            res = response;
        }).catch(error => {
            res = error.response.status;
        });
        return res;
    } catch (error) {
        console.log('Error checking for holiday from remote',error);
        return error;
    }
}
