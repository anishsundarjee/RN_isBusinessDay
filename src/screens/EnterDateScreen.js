import React from 'react'
import { StyleSheet, Platform, SafeAreaView, View, Dimensions, TouchableOpacity, Text } from 'react-native'
import { Headline, Card, Subheading, Button } from 'react-native-paper';
import { showMessage } from "react-native-flash-message";
import axios from 'axios';
import DateTimePicker from '@react-native-community/datetimepicker';
import Ionicons from 'react-native-vector-icons/Ionicons'
import Spinner from 'react-native-loading-spinner-overlay';
import Colors from '../constants/Colors';

const { width } = Dimensions.get('window');
const ENDPOINT = 'https://date.nager.at/api/v3/publicholidays/'
const COUNTRY = 'AU'

const EnterDateScreen = ({navigation}) => {
    const [isLoading, setIsLoading] = React.useState(false);
    const [showDatePicker, setShowDatePicker] = React.useState(false);
    const [displayDate, setDisplayDate] = React.useState(null);
    const [date, setDate] = React.useState(new Date());

    const _renderDatePicker = () =>{
        if(Platform.OS==='android'){
            if(showDatePicker){
                return (
                        <DateTimePicker
                            value={date}
                            mode="date"
                            display="default"
                            maximumDate={new Date(2050, 0, 1)}
                            minimumDate={new Date(1970, 0, 2)}
                            onChange={ async (e, d) => {
                                setShowDatePicker(false);
                                if(e.type == "set") {
                                    setDate(d);
                                    setDisplayDate(d.toDateString())
                                }
                            }}
                            style={{ backgroundColor: 'white' }}
                        />
                );
            } else return null;
        }
    }

    const isWeekend = (input) => {
        try {
            let checkDate = new Date(input);
            if(checkDate.getDay() === 6) {
                checkDate = checkDate.setDate(checkDate.getDate() + 2);
            }
            else if(checkDate.getDay() === 0) {
                checkDate = checkDate.setDate(checkDate.getDate() + 1);
            }
            else {
                console.log('check :',checkDate);
            }
            return new Date(checkDate);
        } catch (error) {
            console.log('Error occured checking date for weekend',error)
        }
    }

    const addToDate = (input) => {
        try {
            let checkDate = new Date(input);
            if(checkDate.getDay() === 6) { //saturday
                checkDate = checkDate.setDate(checkDate.getDate() + 2);
            }
            else if(checkDate.getDay() === 5) { //friday
                checkDate = checkDate.setDate(checkDate.getDate() + 3);
            }
            else { //other weekdays or sunday
                checkDate = checkDate.setDate(checkDate.getDate() + 1);
            }
            console.log('updated date is : ',new Date(checkDate));
            return new Date(checkDate);
        } catch (error) {
            console.log('Error occured checking date for weekend',error)
        }
    }

    const _getListOfHolidays = async (input) => {
        try {
            let checkDate = input;
            const year = checkDate.getFullYear();
            let res = null;
            const URL = `${ENDPOINT}${year}/${COUNTRY}`;
            await axios.get(URL,{ timeout:5000,
                headers : {
                'Accept': 'application/json',
                'Content-Type' : 'application/json'
            }})
            .then(response => {
                res = response;
            }).catch(error => {
                res = error.response.status;
            });
            return res;
        } catch (error) {
            console.log('Error checking for holiday from remote',error);
            showMessage({
                message: "No Network",
                description: "Need network connectivity to work",
                type: "custom",
                icon: "custom",
                floating: true,
                duration: 3000
            });
        }
    }

    const _setDateinFormat = (input) => {
        let currentDayOfMonth = input.getDate();
        currentDayOfMonth < 9 ? currentDayOfMonth = `0${currentDayOfMonth}` : currentDayOfMonth

        let currentMonth = input.getMonth() + 1; // note! January is 0, not 1
        currentMonth < 10 ? currentMonth = `0${currentMonth}` : currentMonth

        const currentYear = input.getFullYear();

        return new Date(currentYear + "-" + currentMonth + "-" + currentDayOfMonth);
    }

    const _setOutputDateFormat = (input) => {
        const currentDayOfMonth = input.getDate();
        const currentMonth = input.getMonth() + 1; // note! January is 0, not 1
        const currentYear = input.getFullYear();

        return currentYear + "-" + currentMonth + "-" + currentDayOfMonth
    }

    const _checkDate = async () => {
        try {
            const checkDate = date;
            let weekday = isWeekend(checkDate);
            let dateCheck = _setDateinFormat(weekday);
            let holidayOutput = await _getListOfHolidays(weekday);
            let businessDay = null;
            if(holidayOutput.status === 200) {
                holidayOutput.data.map((obj, index) => {
                    let output = Math.abs(new Date(obj.date)) - Math.abs(dateCheck);
                    if(output === 0) {
                        dateCheck = addToDate(dateCheck);  
                        if(index === holidayOutput.data.length - 1) {
                            businessDay = dateCheck;
                        }
                    }
                    else if(output < 0) {
                        if(index === holidayOutput.data.length - 1) {
                            businessDay = dateCheck;
                        }
                    }
                    else {
                        businessDay = dateCheck;
                    }
                    // console.log('here we go ::',output, new Date(obj.date), dateCheck, index);
                });
            } else if(holidayOutput.status === 204){
                businessDay = dateCheck;
            } else {
                showMessage({
                    message: "Something went wrong",
                    description: "Invalid data was input.",
                    type: "danger",
                    icon: "danger",
                    floating: true,
                });
            }
            navigation.navigate('ViewOutputScreen',{businessDay: _setOutputDateFormat(businessDay)});
        } catch (error) {
            console.log('Error occured finding business date',error);
        }
    }

    const validateBusinessDate = () => {
        try {
            setIsLoading(true)
            if (Object.prototype.toString.call(date) === "[object Date]") {
                // it is a date
                _checkDate();
                if (isNaN(date.getTime())) {
                    // date is not valid
                    showMessage({
                        message: "Invalid Input",
                        description: "Invalid data was input.",
                        type: "danger",
                        icon: "danger",
                        floating: true,
                    });
                } else {
                    // date is valid
                    _checkDate();
                }
            } else {
                // not a date
                showMessage({
                    message: "Invalid Input",
                    description: "Invalid data was input.",
                    type: "danger",
                    icon: "danger",
                    floating: true,
                });
            }
            setIsLoading(false);
        } catch (error) {
            setIsLoading(false);
            console.log('Error occured validating business date format',error);
        }
    }

    return (
        <SafeAreaView style={styles.container}>
            <Card style={styles.cardStyle}>
                <View style={styles.headerViewStyle}>
                    <Headline style={styles.headerTextStyle}>
                        Welcome to Business Day Checker
                    </Headline>
                </View>
                <View style={styles.wrapperViewStyle}>
                    <View style={styles.subHeadingViewStyle}>
                        <Subheading style={styles.subHeadingTextStyle}>
                            Choose date from below date picker
                        </Subheading>
                    </View>
                    <View style= {styles.DatePickerViewStyle}>
                        <Text style={styles.DatePickerLableStyle}>Business Date</Text>
                        { Platform.OS == "ios" ? (
                            <DateTimePicker
                                placeholderText="Enter date"
                                value={date}
                                mode="date"
                                display="compact"
                                onChange={ async (e, d) => {
                                    // console.log(d);
                                    setDate(d);
                                    setDisplayDate(d.toDateString());
                                }}
                                maximumDate={new Date(2050, 0, 1)}
                                minimumDate={new Date(1970, 0, 2)}
                                style={styles.DatePickerButtonStyle}
                            />
                        ):(
                            <TouchableOpacity 
                                onPress={()=>setShowDatePicker(true)}
                                style={styles.DatePickerButtonStyle}
                            >
                                <Ionicons name="calendar" size={28} color={Colors.PRIMARY_COLOR} style={{marginHorizontal: 10}}/>
                                <Text style={styles.DatePickerTextStyle}>
                                    {displayDate == null || displayDate.length == 0 ? "Tap to enter date." : _setOutputDateFormat(date)}
                                </Text>
                            </TouchableOpacity>
                        )}
                    </View>
                    <Button
                        icon="briefcase-edit"
                        mode="contained"
                        onPress={() => validateBusinessDate()}
                        color={Colors.HINT_COLOR}
                        style={{width: width - 50, alignSelf:'center', marginTop: 20}}
                        disabled={isLoading}
                    >
                        Check Date
                    </Button>
                </View>
                <Spinner
                    visible={isLoading}
                    textContent={'Loading...'}
                    textStyle={styles.spinnerTextStyle}
                />
            </Card>
            {_renderDatePicker()}
        </SafeAreaView>
    )
};

export default EnterDateScreen;

const styles = StyleSheet.create({
    container: {
        flex:1,
        backgroundColor: Colors.WHITE,
    },
    spinnerTextStyle: {
        color: Colors.PRIMARY_COLOR
      },
    cardStyle: {
        backgroundColor: Colors.PRIMARY_BG_COLOR,
        flex:1,
        margin: 10,
        shadowColor: Colors.ACCENT_COLOR,
        shadowOpacity: 5,
        elevation: 4,
        borderRadius: 10,
    },
    headerViewStyle: {
        flex:1,
        alignSelf:'center',
        marginVertical: 20,
        marginHorizontal: 10,
    },
    headerTextStyle: {
        fontSize: 24,
        color: Colors.HINT_COLOR,
        letterSpacing: 1,
        textAlign:'center',
        fontWeight: '600',
    },
    subHeadingTextStyle: {
        fontSize: 16,
        color: Colors.ACCENT_COLOR,
        letterSpacing: 1,
        textAlign:'center',
        fontWeight: '400',
    },
    subHeadingViewStyle: {
        alignSelf:'center',
        marginHorizontal: 10,
    },
    center: {
        justifyContent: 'center',
        alignItems:'center',
    },
    DatePickerViewStyle : {
        flexDirection: 'column', 
        alignSelf:'center', 
        width: width - 30, 
        justifyContent:'center', 
        marginVertical: 5,
        left: 10 
    },
    DatePickerLableStyle: {
        fontSize: 16, 
        fontWeight: '600', 
        color: Colors.ACCENT_COLOR, 
        paddingVertical: 5
    },
    DatePickerButtonStyle: {
        width: width - 50, 
        height: 50, 
        borderWidth: 1, 
        borderRadius:10, 
        borderColor: Colors.TEXTBOX_BORDER, 
        flexDirection: 'row', 
        alignItems: 'center'
    },
    DatePickerTextStyle: {
        alignSelf:'center', 
        fontSize: 16, 
        fontWeight: '500', 
        color: Colors.ACCENT_COLOR
    },
    wrapperViewStyle: {
        width: width - 20,
        flex:3,
        alignSelf:'center',
        justifyContent:'flex-start',
    },
});
