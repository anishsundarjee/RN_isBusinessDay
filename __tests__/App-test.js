/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import EnterDateScreen from '../src/screens/EnterDateScreen'
import { _setDateinFormat, _setOutputDateFormat, addToDate, isWeekend, _getListOfHolidays} from '../src/components/businessLogic'
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {render, fireEvent, waitFor} from '@testing-library/react-native';
global.fetch = require('jest-fetch-mock');

it('renders correctly', () => {
	renderer.create(<App />);
});

test('EnterDateScreen renders correctly', () => {
	const tree = renderer.create(<EnterDateScreen />).toJSON();
	expect(tree).toMatchSnapshot();
});

describe('Checking isWeekend Func', ()=>{
	it('Check wether entered date is a weekend when date entered is weekend', ()=>{
		expect(isWeekend(new Date(`2022-01-08`)))
		.toEqual(new Date(`2022-01-10`))
	});

	it('Check wether entered date is a weekend when date entered is weekday', ()=>{
		expect(isWeekend(new Date(`2022-01-07`)))
		.toEqual(new Date(`2022-01-07`))
	});

	it('Check when invalid date is entered', ()=>{
		expect(isWeekend(new Date('banana')))
		.toBeUndefined()
	});
});

describe('Add to date when there is a public holiday', ()=>{
	it('Check wether entered date is a weekend when date entered is weekend', ()=>{
		expect(addToDate(new Date(`2022-01-08`)))
		.toEqual(new Date(`2022-01-10`))
	});

	it('Check wether entered date is a weekend when date entered is weekday', ()=>{
		expect(addToDate(new Date(`2022-01-10`)))
		.toEqual(new Date(`2022-01-11`))
	});

	it('Check when invalid date is entered', ()=>{
		expect(addToDate(new Date('banana')))
		.toBeUndefined()
	});
});

describe('When business date is found, parse date in correct format', ()=>{
	it('Check if date in required format is returned', ()=>{
		expect(_setOutputDateFormat(new Date(`2022-01-07`)))
		.toEqual(`2022-1-7`)
	});

	it('Check when invalid date is entered', ()=>{
		expect(_setOutputDateFormat(new Date('banana')))
		.toBeUndefined()
	});
});

describe('Making sure date is without any time for strict time comparision', ()=>{
	it('Check if date in required format is returned', ()=>{
		expect(_setDateinFormat(new Date(`2022-01-07`)))
		.toEqual(new Date(`2022-01-07`))
	});

	it('Check when invalid date is entered', ()=>{
		expect(_setDateinFormat(new Date('banana')))
		.toBeUndefined()
	});
});

beforeEach(() => {
	fetch.resetMocks();
})

test('returns 1st object data as new years day 01/01/year', () => {
	const onResponse = jest.fn();
	const onError = jest.fn();
  
	return _getListOfHolidays('2020')
	  .then(onResponse)
	  .catch(onError)
	  .finally(() => {
		// console.log('it is ::',onResponse.mock.calls[0][0].data[0].date);
		expect(onResponse).toHaveBeenCalled();
		expect(onError).not.toHaveBeenCalled();
		expect(onResponse.mock.calls[0][0].data[0].date).toEqual('2020-01-01');
	  });
});

test('returns error when year is invalid input', () => {
	const onResponse = jest.fn();
	const onError = jest.fn();
  
	return _getListOfHolidays('apple')
	  .then(onResponse)
	  .catch(onError)
	  .finally(() => {
		console.log('it is ::',onResponse.mock.results[0].value);
		expect(onResponse).toHaveBeenCalled();
		expect(onError).not.toHaveBeenCalled();
		expect(onResponse.mock.results[0].value).toBeUndefined();
	  });
});
