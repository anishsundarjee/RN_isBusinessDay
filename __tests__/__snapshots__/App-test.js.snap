// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterDateScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "borderRadius": 10,
        "elevation": 4,
        "flex": 1,
        "margin": 10,
        "shadowColor": "#4a4d4e",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 5,
        "shadowRadius": 4,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        index={0}
        siblings={
          Array [
            "View",
            "View",
          ]
        }
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "marginHorizontal": 10,
            "marginVertical": 20,
          }
        }
        total={3}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 24,
                    "letterSpacing": 0,
                    "lineHeight": 32,
                    "marginVertical": 2,
                  },
                  Object {
                    "color": "#1D3540",
                    "fontSize": 24,
                    "fontWeight": "600",
                    "letterSpacing": 1,
                    "textAlign": "center",
                  },
                ],
              ],
            ]
          }
        >
          Welcome to Business Day Checker
        </Text>
      </View>
      <View
        index={1}
        siblings={
          Array [
            "View",
            "View",
          ]
        }
        style={
          Object {
            "alignSelf": "center",
            "flex": 3,
            "justifyContent": "flex-start",
            "width": 730,
          }
        }
        total={3}
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "marginHorizontal": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                      "letterSpacing": 0.5,
                      "lineHeight": 24,
                      "marginVertical": 2,
                    },
                    Object {
                      "color": "#4a4d4e",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "letterSpacing": 1,
                      "textAlign": "center",
                    },
                  ],
                ],
              ]
            }
          >
            Choose date from below date picker
          </Text>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "column",
              "justifyContent": "center",
              "left": 10,
              "marginVertical": 5,
              "width": 720,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#4a4d4e",
                "fontSize": 16,
                "fontWeight": "600",
                "paddingVertical": 5,
              }
            }
          >
            Business Date
          </Text>
        </View>
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#1D3540",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0,
              "elevation": 2,
              "marginTop": 20,
              "minWidth": 64,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0.75,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 1.5,
              "width": 700,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
            testID="submit-button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "marginLeft": 12,
                    "marginRight": -4,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "fontSize": 16,
                      },
                      Array [
                        Object {
                          "lineHeight": 16,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ó°ª˜
                </Text>
              </View>
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Check Date
              </Text>
            </View>
          </View>
        </View>
      </View>
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        statusBarTranslucent={true}
        supportedOrientations={
          Array [
            "landscape",
            "portrait",
          ]
        }
        transparent={true}
        visible={false}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "flex": 1,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.25)",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <ActivityIndicator
              color="white"
              size="large"
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  Object {},
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "flex": 1,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 20,
                      "fontWeight": "bold",
                      "height": 50,
                      "top": 80,
                    },
                    Object {
                      "color": "#E60000",
                    },
                  ]
                }
              >
                Loading...
              </Text>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
</RCTSafeAreaView>
`;
